k8s/buildkite/agent-deployment.yaml

```diff
@@ -19,6 +19,10 @@ spec:
         - name: agent
           image: buildkite/agent:edge-alpine-k8s
           env:
+            - name: BUILDKITE_AGENT_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
```

Comment:
Oh this valueFrom fieldRef stuff is neat!

Reply:
Yeah it's all from [Downward API](https://kubernetes.io/docs/concepts/workloads/pods/downward-api/)